FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1494 @@
+ */
+#define COMMAND_QUEUE_SIZE                  25
+
+/**
+ * @brief Max number of received publishes that can be enqueued for a task.
+ */
+#define PUBLISH_QUEUE_SIZE                  20
+
+/**
+ * @brief Delay for the subscriber task when no publishes are waiting in the queue.
+ */
+#define SUBSCRIBE_TASK_DELAY_MS             400U
+
+/**
+ * @brief Delay for the publisher task between synchronous publishes.
+ */
+#define PUBLISH_DELAY_SYNC_MS               500U
+
+/**
+ * @brief Delay for the publisher task between asynchronous publishes.
+ */
+#define PUBLISH_DELAY_ASYNC_MS              50U
+
+/**
+ * @brief Notification bit indicating completion of publisher task.
+ */
+#define PUBLISHER_TASK_COMPLETE_BIT         ( 1U << 1 )
+
+/**
+ * @brief Notification bit indicating completion of subscriber task.
+ */
+#define SUBSCRIBE_TASK_COMPLETE_BIT         ( 1U << 2 )
+
+/**
+ * @brief Notification bit used by subscriber task for subscribe operation.
+ */
+#define SUBSCRIBE_COMPLETE_BIT              ( 1U << 0 )
+
+/**
+ * @brief Notification bit used by subscriber task for unsubscribe operation.
+ */
+#define UNSUBSCRIBE_COMPLETE_BIT            ( 1U << 1 )
+
+/**
+ * @brief Maximum number of loop iterations to wait for a task notification.
+ */
+#define MAX_WAIT_ITERATIONS                 5
+
+/**
+ * @brief Topic filter used by the subscriber task.
+ */
+#define SUBSCRIBE_TOPIC_FILTER              "publish/+/filter"
+
+/**
+ * @brief Format string used by the publisher task for topic names.
+ */
+#define PUBLISH_TOPIC_FORMAT_STRING         "publish/%i/filter"
+
+/**
+ * @brief Format string used by the publisher task for payloads.
+ */
+#define PUBLISH_PAYLOAD_FORMAT              "Hello World! %d"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief A type of command for interacting with the MQTT API.
+ */
+typedef enum CommandType
+{
+    PROCESSLOOP, /**< @brief Call MQTT_ProcessLoop(). */
+    PUBLISH,     /**< @brief Call MQTT_Publish(). */
+    SUBSCRIBE,   /**< @brief Call MQTT_Subscribe(). */
+    UNSUBSCRIBE, /**< @brief Call MQTT_Unsubscribe(). */
+    PING,        /**< @brief Call MQTT_Pint(). */
+    DISCONNECT,  /**< @brief Call MQTT_Disconnect(). */
+    CONNECT,     /**< @brief Placeholder command for reconnecting a broken connection. */
+    TERMINATE    /**< @brief Exit the command loop and stop processing commands. */
+} CommandType_t;
+
+/**
+ * @brief Struct containing context for a specific command.
+ *
+ * @note An instance of this struct and any variables it points to MUST stay
+ * in scope until the associated command is processed, and its callback called.
+ * The command callback will set the `complete` flag, and notify the calling task.
+ *
+ * @note This struct does not accept a parameter for number of subscriptions.
+ * Only a single subscription per SUBSCRIBE command is allowed in this demo.
+ * This is to simplify subscription management.
```

Comment:
I think it will be a good addition to mention about what needs to be done to support multiple subscriptions in a command to extend the demo if needed by customers.

Reply:
Forgot to remove this, thanks